cdap-app-fabric/src/main/java/io/cdap/cdap/internal/credential/handler/CredentialProviderHttpHandler.java

```diff
@@ -0,0 +1,409 @@
+        createRequest.getName());
+    final CredentialIdentity identity = createRequest.getIdentity();
+    if (createRequest.getIdentity() == null) {
+      throw new BadRequestException("No identity provided for create request");
+    }
+    accessEnforcer.enforce(identityId, StandardPermission.CREATE);
+    ensureNamespaceExists(namespace);
+    validateIdentity(identity);
+    accessEnforcer.enforce(identity.getCredentialProfile(), StandardPermission.GET);
+    credentialIdentityManager.create(identityId, identity);
+    responder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  /**
+   * Updates an existing identity.
+   *
+   * @param request The HTTP request.
+   * @param responder The HTTP responder.
+   * @param namespace The identity namespace.
+   * @param identityName The identity name.
+   * @throws BadRequestException If the identity name or identity are invalid.
+   * @throws IOException If transport errors occur.
+   * @throws NotFoundException If the namespace or identity are not found.
+   */
+  @PUT
+  @Path("/namespaces/{namespace-id}/credentials/identities/{identity-name}")
+  public void updateIdentity(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace,
+      @PathParam("identity-name") String identityName)
+      throws BadRequestException, IOException, NotFoundException {
+    final CredentialIdentityId identityId = createIdentityIdOrPropagate(namespace, identityName);
+    accessEnforcer.enforce(new CredentialIdentityId(namespace, identityName),
+        StandardPermission.UPDATE);
+    ensureNamespaceExists(namespace);
+    final CredentialIdentity identity = deserializeRequestContent(request, CredentialIdentity.class);
+    validateIdentity(identity);
+    accessEnforcer.enforce(identity.getCredentialProfile(), StandardPermission.GET);
+    credentialIdentityManager.update(identityId, identity);
+    responder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  /**
+   * Deletes an identity.
+   *
+   * @param request The HTTP request.
+   * @param responder The HTTP responder.
+   * @param namespace The identity namespace.
+   * @param identityName The identity name.
+   * @throws BadRequestException If the identity name is invalid.
+   * @throws IOException If transport errors occur.
+   * @throws NotFoundException If the namespace or identity are not found.
+   */
+  @DELETE
+  @Path("/namespaces/{namespace-id}/credentials/identities/{identity-name}")
+  public void deleteIdentity(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespace,
+      @PathParam("identity-name") String identityName)
+      throws BadRequestException, IOException, NotFoundException {
+    final CredentialIdentityId identityId = createIdentityIdOrPropagate(namespace, identityName);
+    accessEnforcer.enforce(new CredentialIdentityId(namespace, identityName),
+        StandardPermission.DELETE);
+    ensureNamespaceExists(namespace);
```

Comment:
Is it intentional that creating and updating identities requires GET permissions on the credential profile, but delete does not?

Reply:
Yup, because when you create or update an identity, you are associating it with a particular profile. We might want to change the GET permission check to USE in a future PR too, depending on what's necessary to use the identity to provision a credential.